src/components/EditCoursePage/EditCourseForm.test.jsx

```diff
@@ -244,6 +246,50 @@ describe('BaseEditCourseForm', () => {
     expect(disabledFields).toHaveLength(1);
   });
 
+  it('check for customValidity working correctly for url_slug in EditCourseForm', () => {
+    const setCustomValidityMock = jest.fn();
+    const courseInfoWithUrlSlug = {
+      data: {
+        url_slug: 'test-url-slug/',
+      },
+    };
+    const initialValuesWithUrlSlug = {
+      ...initialValuesFull,
+      course_runs: [],
+    };
+    const component = shallow(
+      <BaseEditCourseForm
+        handleSubmit={() => null}
+        title={initialValuesWithUrlSlug.title}
+        initialValues={initialValuesWithUrlSlug}
+        currentFormValues={initialValuesWithUrlSlug}
+        number="Test101x"
+        entitlement={{ sku: 'ABC1234' }}
+        courseStatuses={[PUBLISHED]}
+        courseInfo={courseInfoWithUrlSlug}
+        courseOptions={courseOptions}
+        courseRunOptions={courseRunOptions}
+        uuid={initialValuesWithUrlSlug.uuid}
+        type={initialValuesWithUrlSlug.type}
+        id="edit-course-form"
+      />,
+    );
+
+    const urlSlugField = component.find({ name: 'url_slug' });
+
+    // Invalid input
+    const invalidInput = 'Invalid-URL-Slug123/';
+    urlSlugField.simulate('input', { target: { value: invalidInput } });
+    urlSlugField.prop('extraInput').onInvalid({ target: { setCustomValidity: setCustomValidityMock } });
```

Comment:
Why is this mocking added here? Shouldn't this be before simulating input?

Reply:
Doesn't matter because onInvalid method is called on form submission. 
fixed. I've moved it to above line.